ui/src/app/connect/dialog/edit-event-property/edit-event-property.component.ts

```diff
@@ -33,159 +44,215 @@ import { EditValueTransformationComponent } from './components/edit-value-transf
+            (result as any).timestampTransformationFormatString = (
+                ep as any
+            ).timestampTransformationFormatString;
+            (result as any).timestampTransformationMultiplier = (
+                ep as any
+            ).timestampTransformationMultiplier;
+
+            (result as any).staticValue = (ep as any).staticValue;
+
+            (result as any).correctionValue = (ep as any).correctionValue;
+            (result as any).operator = (ep as any).operator;
+
+            return result;
+        } else if (ep instanceof EventPropertyNested) {
+            return EventPropertyNested.fromData(
+                ep as EventPropertyNested,
+                new EventPropertyNested(),
+            );
+        } else {
+            return EventPropertyList.fromData(
+                ep as EventPropertyList,
+                new EventPropertyList(),
+            );
+        }
+    }
 
-      (this.property as any).measurementUnitTmp = (this.cachedProperty as any).measurementUnitTmp;
-      (this.property as any).oldMeasurementUnit = (this.cachedProperty as any).oldMeasurementUnit;
-      (this.property as any).hadMeasarumentUnit = (this.cachedProperty as any).hadMeasarumentUnit;
+    private createForm() {
+        this.propertyForm = this.formBuilder.group({
+            label: [this.property.label, Validators.required],
+            runtimeName: [this.property.runtimeName, Validators.required],
+            description: [this.property.description, Validators.required],
+            domainProperty: ['', Validators.required],
+            dataType: ['', Validators.required],
+        });
+    }
 
-      (this.property as any).timestampTransformationMode = (this.cachedProperty as any).timestampTransformationMode;
-      (this.property as any).timestampTransformationFormatString = (this.cachedProperty as any).timestampTransformationFormatString;
-      (this.property as any).timestampTransformationMultiplier = (this.cachedProperty as any).timestampTransformationMultiplier;
+    staticValueAddedByUser() {
+        return this.property.elementId.startsWith(
+            'http://eventProperty.de/staticValue/',
+        );
+    }
 
-      (this.property as any).staticValue = (this.cachedProperty as any).staticValue;
+    save(): void {
+        this.property.label = this.cachedProperty.label;
+        this.property.description = this.cachedProperty.description;
+
+        // remove undefined from domain properies array
+        this.property.domainProperties =
+            this.cachedProperty.domainProperties.filter(n => n);
+        this.property.runtimeName = this.cachedProperty.runtimeName;
+        this.property.propertyScope = this.cachedProperty.propertyScope;
+
+        if (this.property instanceof EventPropertyList) {
+            // @ts-ignore
+            this.property.eventProperty.runtimeType = (
+                this.cachedProperty as EventPropertyList
+            ).eventProperty.runtimeType;
```

Comment:
cypress-e2e-tests fails when building the UI

file: `ui/src/app/connect/dialog/edit-event-property/edit-event-property.component.ts",`

```
Property 'runtimeType' does not exist on type 'EventPropertyUnion'.
  Property 'runtimeType' does not exist on type 'EventPropertyList'.
```

![image](https://user-images.githubusercontent.com/56036364/211169355-914cf3a3-ce6d-4f1f-a974-747c4851812d.png)

It seems like, inlining the code fixes the issue. After Inlining, not getting any errors.

![image](https://user-images.githubusercontent.com/56036364/211169598-0f38694c-b72b-4749-a195-c973d4f99969.png)

Any thoughts? @bossenti

Reply:
Can you try to remove the @ts-ignore comment?

Reply:
I have tried removing it. Still getting the same error.

Reply:
For me the second one is more intuitive/easier to read
What do you think @tenthe or @dominikriemer?

Reply:
Hey @aamirxshaikh,
I had a look at the code and you can remove the whole block:
```Typescript
if (this.property instanceof EventPropertyList) {
    // @ts-ignore
    this.property.eventProperty.runtimeType = (
        this.cachedProperty as EventPropertyList
    ).eventProperty.runtimeType;
}
```
Since the value is not part of every `EventProperty` object the value is ignored anyway.